### PR TITLE
fix(hybridcloud) Add retries to sync webhook operations

### PR DESCRIPTION
**Pull request by markstory**
Some integrations (jira issue glance) require synchronous operations during proxied requests. These operations are delegated to the region from control, and during application restarts we can hit 503s. Having retires will help smooth over those blips.

Fixes SENTRY-2DKM

### CODE REVIEW

File: src/sentry/options/defaults.py [hunk 1]
```diff
@@ -1644,8 +1644,9 @@
 register("hybrid_cloud.region-user-allow-list", default=[], flags=FLAG_AUTOMATOR_MODIFIABLE)
 
 # Retry controls
-register("hybrid_cloud.rpc.retries", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
-register("hybrid_cloud.integrationproxy.retries", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
+register("hybridcloud.regionsiloclient.retries", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
+register("hybridcloud.rpc.retries", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
+register("hybridcloud.integrationproxy.retries", default=5, flags=FLAG_AUTOMATOR_MODIFIABLE)
```
Comment (markstory) [inline src/sentry/options/defaults.py:1649]: Aligned the names of these options with the eventual module names. None of these options have been customized yet.

File: src/sentry/middleware/integrations/parsers/jira.py [hunk 2]
```diff
@@ -73,7 +74,11 @@ def get_response(self):
             )
 
         if self.view_class in self.immediate_response_region_classes:
-            return self.get_response_from_region_silo(region=regions[0])
+            try:
+                return self.get_response_from_region_silo(region=regions[0])
+            except ApiError as err:
```
Comment (sentaur-athena) [inline src/sentry/middleware/integrations/parsers/jira.py:79]: What kind of error are expecting here that getting response from control would fix it?

File: src/sentry/middleware/integrations/parsers/jira.py [hunk 3]
```diff
@@ -73,7 +74,11 @@ def get_response(self):
...
+                sentry_sdk.capture_exception(err)
+                return self.get_response_from_control_silo()
```
Comment (sentaur-athena) [inline src/sentry/middleware/integrations/parsers/jira.py:81]: Isn't this covered in line 63? I guess same as my previous question.

**Review by sentaur-athena — COMMENTED** [review]
Just a few questions to understand better

File: src/sentry/middleware/integrations/parsers/jira.py (hunk 2 above)
Comment (markstory) [reply src/sentry/middleware/integrations/parsers/jira.py:79]: We get an `ApiRetryError` if all retries are exhausted.

File: src/sentry/middleware/integrations/parsers/jira.py (hunk 3 above)
Comment (markstory) [reply src/sentry/middleware/integrations/parsers/jira.py:81]: If we weren't able to get a response from the region due to timeouts/retry attempts being exceeded, I didn't want to let the exception bubble up as the response would be a 500. We could raise another an Http Exception that won't be captured as an error here as well.

**Review by sentaur-athena — APPROVED** [review]

File: src/sentry/silo/client.py [hunk 4]
```diff
@@ -176,21 +178,35 @@ class RegionSiloClient(BaseSiloClient):
...
+
+        return build_session(
+            is_ipaddress_permitted=validate_region_ip_address,
+            max_retries=Retry(
+                total=options.get("hybridcloud.regionsiloclient.retries"),
+                backoff_factor=0.1,
+                status_forcelist=[503],
+                allowed_methods=["PATCH", "HEAD", "PUT", "GET", "DELETE", "POST"],
+            ),
+        )
```
Comment (ykamo001) [inline src/sentry/silo/client.py:209]: is this a synchronous backoff or async?

Comment (markstory) [reply src/sentry/silo/client.py:209]: It is a synchronous backoff.

**Pull request merged**